src/objects/objects.cc

```diff
@@ -5543,21 +5543,16 @@ Handle<Object> JSPromise::Reject(Handle<JSPromise> promise,
   // 6. Set promise.[[PromiseState]] to "rejected".
   promise->set_status(Promise::kRejected);
 
-  {
-    v8::replayio::AutoMaybeDisallowEvents disallow(promise->has_handler(),
-                                                   "[TT-1029-1030] JSPromise::Reject %d");
```

Comment:
I removed this AutoMaybeDisallowEvents because in the associated issue it was a speculative change for a mismatch which I believe is fixed by the change in this PR to RecordReplayShouldCallOnPromiseHook.

We want promise->has_handler() to be consistent between recording and replaying, so conditionally disallowing events for promises which have handlers is unnecessary and confusing and potentially masks effects which mismatched behavior would have.